apps/global_full/4C_global_full_main.cpp

```diff
@@ -343,14 +344,35 @@ int main(int argc, char* argv[])
       input_file.emit_metadata(std::cout);
     }
   }
-  else if ((argc == 4) && (strcmp(argv[1], "--to-yaml") == 0))
+  else if ((argc >= 3) && (strcmp(argv[1], "--to-yaml") == 0))
   {
     if (Core::Communication::my_mpi_rank(lcomm) == 0)
     {
+      std::filesystem::path inputfile_name = argv[2];
+      std::filesystem::path outputfile_name;
       Core::IO::InputFile input_file = Global::set_up_input_file(lcomm);
-      input_file.read(argv[2]);
-      std::ofstream output_file(argv[3]);
+      input_file.read(inputfile_name);
+      if (argc == 3)
+      {
+        outputfile_name = inputfile_name.replace_extension("yaml");
+        if (std::filesystem::exists(outputfile_name))
+        {
+          printf("You did not provide an output file name.\n");
+          printf("The default is to replace the original suffix by .yaml.\n");
+          printf("However, the file '%s' already exists. I will not continue\n",
+              outputfile_name.c_str());
+          MPI_Finalize();
```

Comment:
```suggestion
```
I think, this is not necessary as `MPI_Finalize` will be called by the `struct CleanUpMPI` in the destructor.